ospf6d/ospf6_nssa.c

```diff
@@ -1095,6 +1095,33 @@ static void ospf6_check_and_originate_type7_lsa(struct ospf6_area *area)
 
 }
 
+static void ospf6_nssa_update_timer(struct thread *thread)
+{
+	struct ospf6 *ospf6 = THREAD_ARG(thread);
+	struct ospf6_area *area;
+	struct listnode *lnode, *nnode;
+
+	if (IS_OSPF6_DEBUG_NSSA)
+		zlog_debug("%s: Start", __func__);
+
+	for (ALL_LIST_ELEMENTS(ospf6->area_list, lnode, nnode, area)) {
```

Comment:
This loop could be done with `ALL_LIST_ELEMENTS_RO` since the `ospf6->area_list` is not modified.